File: java/1431/Solution.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> answer = new ArrayList<>();

        int maxCandy = Arrays.stream(candies)
            .max()
            .getAsInt();

        int minRequiredCandy = maxCandy - extraCandies;

        for (int candy : candies) {
            answer.add(candy >= minRequiredCandy);
        }

        return answer;
    }
}

File: python3/1431/Solution.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        answer = []

        min_required_candy = max(candies) - extraCandies

        for candy in candies:
            answer.append(candy >= min_required_candy)

        return answer
